src/main/java/uk/gov/ch/oauth/IdentityProvider.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.ch.oauth;
+
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+public class IdentityProvider implements IIdentityProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("docs.developer.ch.gov.uk");
+    private final static Decoder decoder = Base64.getDecoder();
+    final private byte[] requestKey;
+    final private String authorizationUri;
+    final private String clientId;
+    final private String redirectUri;
+    final private String clientSecret;
+    final private String tokenUrl = null;
+    final private String profileUrl = null;
+    @Value("${home.url}")
+    private String homeUrl;
+
+    @Autowired
+    public IdentityProvider(final EnvironmentReader reader) {
+        requestKey = decoder.decode(reader.getMandatoryString("DEVELOPER_OAUTH2_REQUEST_KEY"));
+        this.authorizationUri = reader.getMandatoryString("OAUTH2_AUTH_URI");
+        this.clientId = reader.getMandatoryString("CHS_DEVELOPER_CLIENT_ID");
+        clientSecret = reader.getMandatoryString("CHS_DEVELOPER_CLIENT_SECRET");
+        redirectUri = reader.getMandatoryString("OAUTH2_REDIRECT_URI");
+        // TODO is this better? Which is the correct source of the redirect URL? redirectUri = reader.getMandatoryString("REDIRECT_URI");
```

Comment:
Do we still want this comment if we've re-implemented the code?

Reply:
This is really a question of where is the source of truth for this value. I could get the OAUTH2... variant from the existing environment.

Reply:
Do we have a plan for resolving the question and an understanding of how to fix this in future?

Reply:
I think a decision needs to be made. My preference would be for an environment variable named so that it's obvious for OAUTH usage.

Reply:
See my comment above, just do a pull on vagrant and you should see this value.

Reply:
I have uncommented the line out and changed the var to redirectUriPage - which only deals with the redirect to the home page.